test/e2e/tls_test.go

```diff
@@ -131,27 +134,30 @@ func (data *TestData) opensslConnect(t *testing.T, pod string, container string,
 	}{
 		{
 			clusterInfo.podV4NetworkCIDR != "",
-			"127.0.0.1",
+			nodeIPv4,
 			"-4",
 		},
 		{
 			clusterInfo.podV6NetworkCIDR != "",
-			"::1",
+			nodeIPv6,
 			"-6",
 		},
 	}
 	for _, c := range opensslConnectCommands {
 		if !c.enabled {
 			continue
 		}
-		cmd := []string{"timeout", "1", "openssl", "s_client", "-connect", net.JoinHostPort(c.ip, fmt.Sprint(port)), c.option}
+		cmd := []string{"curl", "-k", "-v", fmt.Sprintf("https://%s", net.JoinHostPort(c.ip, fmt.Sprint(port)))}
```

Comment:
nit: it probably makes sense to use `--head` in the request, as we don't care about the response payload